libs/coin-modules/coin-multiversx/src/buildTransaction.ts

```diff
@@ -33,11 +33,56 @@ export const buildTransactionToSign = async (
     sender: address,
     gasPrice: GAS_PRICE,
     gasLimit: transaction.gasLimit || networkConfig.MinGasLimit.valueOf(),
-    data: transaction.data,
     chainID,
     ...HASH_TRANSACTION,
   };
 
+  if (transaction.data) {
+    unsigned.data = transaction.data;
+  }
+
   // Will likely be a call to Elrond SDK
   return JSON.stringify(unsigned);
 };
+
+export const serializeESDTInfo = (
+  ticker: string,
+  id: string,
+  decimals: number,
+  chainId: string,
+  signature: string,
+): Buffer => {
+  const tickerLengthBuffer = Buffer.from([ticker.length]);
+  const tickerBuffer = Buffer.from(ticker);
+  const idLengthBuffer = Buffer.from([id.length]);
+  const idBuffer = Buffer.from(id);
+  const decimalsBuffer = Buffer.from([decimals]);
+  const chainIdLengthBuffer = Buffer.from([chainId.length]);
+  const chainIdBuffer = Buffer.from(chainId);
+  const signatureBuffer = Buffer.from(signature, "hex");
+  const infoBuffer = [
+    tickerLengthBuffer,
+    tickerBuffer,
+    idLengthBuffer,
+    idBuffer,
+    decimalsBuffer,
+    chainIdLengthBuffer,
+    chainIdBuffer,
+    signatureBuffer,
+  ];
+  return Buffer.concat(infoBuffer);
+};
+
+export const provideESDTInfo = async (
+  ticker?: string,
+  id?: string,
+  decimals?: number,
+  chainId?: string,
+  signature?: string,
+): Promise<any> => {
+  if (!ticker || !id || !decimals || !chainId || !signature) {
+    throw new Error("Invalid ESDT token credentials!");
```

Comment:
Perhaps `metadata` instead of `credentials`? However, I think it's existing code, thus can be kept as it is.

Reply:
indeed, it's existing code.